database/prod/query.h

```diff
@@ -24,6 +25,34 @@ class Query : public QSqlQuery {
     QString table_;
 };
 
+class SelectQuery : public Query {
+public:
+    SelectQuery(const QString& table, const std::vector<Field>& columns, const QueryClauses& clauses = QueryClauses());
+
+    void setClauses(const QueryClauses& clauses);
+    void setClause(const BaseClause* clause);
+    void join(const QString& table, const std::pair<Field, Field>& statement);
+    void join(const std::vector<TableAndColumns>& statements);
+    bool exec() override;
+    QString text() const override;
+
+private:
+    class From {
+    public:
+        From(const QString& table);
+        void addJoin(const QString& table, const std::pair<Field, Field>& statement);
+        QString text() const;
+
+    private:
+        QString table_;
+        std::vector<TableAndColumns> joinOn_;
+    };
+
+    const QString SELECT_PART_TEMPLATE = "SELECT\n%1";
```

Comment:
я бы отделила саб-класс от переменных отдельной привейт секцией
как отделяем функции от переменных

```suggestion
private:
    const QString SELECT_PART_TEMPLATE = "SELECT\n%1";
```